doc/null_safety/null_safe_migration.md

```diff
@@ -16,29 +17,76 @@ If you haven't already, familiarize yourself with the concepts of Dart null safe
+dart pub global run over_react_codemod:null_safety_prep --yes-to-all
+dart pub global run over_react_codemod:null_safety_migrator_companion --yes-to-all
 ```
 
-This codemod will:
+These codemods will:
 - Migrate as many over_react specific use cases as possible.
 - Get the repo into a state where the migration tool can be run with less manual intervention.
-
-## Step 3: Run the migration tool
+- Add nullability hints to props/state that are defaulted/initialized in class components.
+  - These hints will cause defaulted/initialized values to be migrated as "late required". 
+    See our [prop nullability](#prop-nullability) docs for more details on whether you should keep them required following the migration.
+
+## Step 3: Enable the analyzer plugin
+
+Enabling the analyzer plugin will help you spot potential null-safety issues in your components at analysis time,
+especially as it pertains to [prop nullability](#prop-nullability).
+
+1. Enable the plugin in your `analysis_options.yaml` file:
+    ```yaml
+    analyzer:
+      plugins:
+        - over_react
+    ```
+1. Restart the Dart Analysis Server in your IDE. The plugin may take a minute to load after built-in analysis completes.
+
+   * If the number of new lints / errors is overwhelming, you an also use this snippet to disable all lints except for the ones that are particularly useful during a null-safety migration:
+       ```yaml
+       analyzer:
+         errors:
+           over_react_exhaustive_deps: ignore
+           over_react_invalid_render_return_type: ignore
+           over_react_pseudo_static_lifecycle: ignore
+           over_react_rules_of_hooks: ignore
+           over_react_style_missing_unit: ignore
+           over_react_avoid_link_target_vulnerability: ignore
+           over_react_boilerplate_warning: ignore
+           over_react_create_ref_usage: ignore
+           over_react_forward_only_dom_props_to_dom_builders: ignore
+           over_react_hash_code_as_key: ignore
+           over_react_invalid_dom_attribute: ignore
+           over_react_low_quality_key: ignore
+           over_react_missing_cascade_parens: ignore
+           over_react_missing_key: ignore
+           over_react_object_to_string_as_key: ignore
+           over_react_proptypes_do_not_throw: ignore
+           over_react_unknown_key_type: ignore
+           over_react_boilerplate_debug: ignore
+           over_react_bool_prop_name_readability: ignore
+           over_react_consumed_props_return_value: ignore
+           over_react_incorrect_doc_comment_location: ignore
+           over_react_prefer_null_over_false_render_return_type: ignore
+           over_react_prefer_use_or_create_ref: ignore
+           over_react_string_ref: ignore
+           over_react_unnecessary_key: ignore
+           over_react_variadic_children: ignore
+         plugins:
+           over_react
+       ```
+
+## Step 4: Run the migration tool
 
 Run the null safety migrator tool:
```

Comment:
It would probably be helpful to remind people to commit hints from the codemod beforehand, so they don't get all mixed up with other hints they add when using the migrator

Reply:
[Clarify that a separate commit is useful](https://github.com/Workiva/over_react/pull/932/commits/e200c1345fe206e559de9b47342321510ef0f007)